GitCommands/Git/GitModule.cs

```diff
@@ -2143,42 +2143,50 @@ public IReadOnlyList<PatchFile> GetInteractiveRebasePatchFiles()
-                var isCurrentFound = false;
-                foreach (string todoCommit in todoCommits)
-                {
-                    if (todoCommit.StartsWith(commentChar))
-                    {
-                        continue;
-                    }
+            string commentChar = EffectiveConfigFile.GetString("core.commentChar", "#");
 
-                    string[] parts = todoCommit.Split(Delimiters.Space);
+            // Filter comment lines and keep only lines containing at least 3 columns
+            // (action, commit hash and commit subject -- that could contains space and be cut in more --)
+            // ex: pick e0d861716540aa1ac83eaa2790ba5e79988b9489 this is the commit subject
+            string[][] todoCommitsInfos = todoCommits.Where(l => !l.StartsWith(commentChar))
+                .Select(l => l.Split(Delimiters.Space))
+                .Where(p => p.Length >= 3)
+                .ToArray();
 
-                    if (parts.Length < 3)
-                    {
-                        continue;
-                    }
+            string? currentCommitShortHash = File.Exists(CurrentFilePath) ? File.ReadAllText(CurrentFilePath).Trim() : null;
+            bool isCurrentFound = false;
 
-                    string commitHash = parts[1];
-                    CommitData? data = _commitDataManager.GetCommitData(commitHash, out var error);
-                    var isApplying = currentCommitShortHash is not null && commitHash.StartsWith(currentCommitShortHash);
-                    isCurrentFound |= isApplying;
+            RevisionReader reader = new(this, hasReflogSelector: false, allBodies: true);
+            CancellationTokenSource cts = new(2 * 60_000);
+            Dictionary<string, GitRevision> rebasedCommitsRevisions =
+                reader.GetRevisionsFromRange(todoCommitsInfos[0][1], todoCommitsInfos[^1][1], cts.Token)
+                .ToDictionary(r => r.Guid, r => r);
 
-                    patchFiles.Add(new PatchFile
-                    {
-                        Author = error ?? data?.Author,
-                        ObjectId = data?.ObjectId,
-                        Subject = error ?? data?.Body,
-                        Action = parts[0],
-                        Date = error ?? data?.CommitDate.LocalDateTime.ToString(),
-                        IsNext = isApplying,
-                        IsApplied = !isCurrentFound,
-                    });
-                }
+            foreach (string[] parts in todoCommitsInfos)
+            {
+                string commitHash = parts[1];
+                CommitData? data = rebasedCommitsRevisions.TryGetValue(commitHash, out GitRevision commitRevision)
+                    ? _commitDataManager.CreateFromRevision(commitRevision, null)
+                    : _commitDataManager.GetCommitData(commitHash, out var _);
+
+                var isApplying = currentCommitShortHash is not null && commitHash.StartsWith(currentCommitShortHash);
```

Comment:
nit (on unmodified code):
```suggestion
                bool isApplying = currentCommitShortHash is not null && commitHash.StartsWith(currentCommitShortHash);
```